membership-common/src/main/scala/com/gu/memsub/subsv2/Subscription.scala

```diff
@@ -76,7 +76,10 @@ object GetCurrentPlans {
       else if (product == Product.Digipack && plan.billingPeriod.toOption.contains(OneTimeChargeBillingPeriod))
         digipackGiftEnded(_ => sub.termEndDate >= dateToCheck)
       else
-        paidPlanEnded(_ => plan.effectiveEndDate >= dateToCheck)
+        paidPlanEnded(_ => {
+          val inGracePeriodAndNotCancelled = plan.effectiveEndDate == dateToCheck && !sub.isCancelled
+          plan.effectiveEndDate > dateToCheck || inGracePeriodAndNotCancelled
+        })
```

Comment:
this was the minimal change, but I have previously had trouble understanding this function, and I'm sure it could be written as a for comprehension or similar.  If I'd had more time I'd have tried to make the whole thing easier to understand.